src/utils/rotationProvider.ts

```diff
@@ -0,0 +1,134 @@
+import { BaseProvider, Network, StaticJsonRpcProvider } from '@ethersproject/providers';
+import { logger } from 'ethers';
+
+const DEFAULT_ROTATION_DELAY = 5000;
+const DEFAULT_FALL_FORWARD_DELAY = 60000;
+
+interface RotationProviderConfig {
+  rotationDelay?: number;
+  fallFowardDelay?: number;
+}
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+/**
+ * Returns the network as long as all agree. Throws an error if any two networks do not match
+ * @param networks the list of networks to verify
+ * @returns Network
+ */
+export function checkNetworks(networks: Network[]): Network {
+  if (networks.length === 0) {
+    logger.throwArgumentError('no networks provided', 'networks', networks);
+  }
+
+  let result: Network | undefined;
+
+  for (let i = 0; i < networks.length; i++) {
+    const network = networks[i];
+
+    if (!network) {
+      logger.throwArgumentError('network not defined', 'networks', networks);
+    }
+
+    if (!result) {
+      result = network;
+      continue;
+    }
+
+    // Make sure the network matches the previous networks
+    if (
+      !(
+        result.name === network.name &&
+        result.chainId === network.chainId &&
+        (result.ensAddress === network.ensAddress ||
+          (result.ensAddress == null && network.ensAddress == null))
+      )
```

Comment:
Do we need normalization of the values here or would they add more robustness to the checks?

Reply:
Updated, normalized name and ensAddress